osu.Game.Rulesets.Taiko/Difficulty/Evaluators/RhythmEvaluator.cs

```diff
@@ -55,10 +67,10 @@ private static double repeatedIntervalPenalty(SameRhythmHitObjects sameRhythmHit
                 ? sameInterval(sameRhythmHitObjects, 4)
                 : 1.0; // Returns a non-penalty if there are 6 or more notes within an interval.
 
-            // Scale penalties dynamically based on hit object duration relative to hitWindow.
-            double penaltyScaling = Math.Max(1 - sameRhythmHitObjects.Duration / (hitWindow * 2), 0.5);
+            // The duration penalty is based on hit object duration relative to hitWindow.
+            double durationPenalty = Math.Max(1 - sameRhythmHitObjects.Duration * 2 / hitWindow, 0.5);
```

Comment:
Was increasing map OD leading to decreased SR fixed?

Reply:
yes, had to switch the logic around in order for it to behave properly.